pkg/api/configsync/v1alpha1/resource_override.go

```diff
@@ -93,34 +93,51 @@ type RootSyncOverrideSpec struct {
 	// +optional
 	NamespaceStrategy configsync.NamespaceStrategy `json:"namespaceStrategy,omitempty"`
 
-	// clusterRole controls which role to bind the service account for this RootSync's
-	// reconciler to.
+	// roleRefs is a list of Roles or ClusterRoles to create bindings.
+	// If unset, a binding to cluster-admin will be created.
 	//
 	// +optional
-	ClusterRole ClusterRoleOverrideSpec `json:"clusterRole,omitempty"`
+	RoleRefs []RootSyncRoleRef `json:"roleRefs,omitempty"`
 }
 
-// ClusterRoleOverrideSpec allows to override settings for the root reconciler permissions
-type ClusterRoleOverrideSpec struct {
-	// name is the name of the ClusterRole to bind to; defaults to "cluster-admin".
-	//
-	// If you specify a different name, the ClusterRole needs to be created manually.
+// RootSyncRoleRef indicates references for Roles and ClusterRoles to create
+// bindings to for this reconciler. It supports a namespace field that can be used
+// to create RoleBindings rather than ClusterRoleBindings.
+type RootSyncRoleRef struct {
+	RoleRefBase `json:",inline"`
+
+	// namespace indicates the Namespace in which a RoleBinding should be created.
+	// For ClusterRole objects, will determine whether a RoleBinding or ClusterRoleBinding
+	// is created.
+	// For Role objects, must be set to the same namespace as the Role.
 	//
 	// +optional
-	Name string `json:"name,omitempty"`
+	Namespace string `json:"namespace,omitempty"`
+}
 
-	// disabled turns off binding to a ClusterRole entirely.
-	//
-	// It is then your responsibility to bind any roles to the relevant service
-	// account.
+// RoleRefBase is the base struct for references to Roles and ClusterRoles. These
+// references are used to create bindings for both RootSyncs and RepoSyncs.
+type RoleRefBase struct {
```

Comment:
If we're not going to add this to RepoSync, you can probably simplify and remove the RoleRefBase

Reply:
Removed.